### DISCUSSION TITLE
Using zod form validation with radio button group that has no default checked value on first render

### DISCUSSION THREAD

**Discussion by sherrellbc** [colinhacks/zod#3640] [Q&A]
I have a simple form that I am using `zod` to validate. For simplicity let's say the form has exactly one user field for specifying a selection:

```
const FormSchema = z.object({
    options: z.enum(["option-0", "option-1", "option-2"], {
        required_error: "You must select an option",
    })
})

export default function Form() {
    const form = useForm < z.infer < typeof FormSchema >> ({
        resolver: zodResolver(FormSchema),
        defaultValues: {
            options: "",
        },
    })

    const onSubmit = useCallback(async (data: z.infer<typeof FormSchema>) => {
        console.log(data)
    }, [])

    return (
            <form onSubmit={form.handleSubmit(onSubmit)} 
            ...
            </form>
    )
}
```

The above code does not work since `""` is not in the set of options specified in the `FormSchema`.

Is there a way to:
 1. Have no radio buttons selected on first render
 2. Still use zod validation _onSubmit_ to ensure the value for the radio button group is within the set specified?

**Comment by lpknv** [comment]
Interesting, this is the exact situation I am in right now.

**Comment by shehabadel** [comment]
I am facing the same issue, but I thought it could be something like this where we union the field with `z.literal("")`

For example,
```typescript
 z.union([
    z.enum(["option-0", "option-1", "option-2"].optional(),
    z.literal(""),
  ])
  ```
